src/VisualStudio/Core/Def/Implementation/ProjectSystem/TextEditApplication.cs

```diff
@@ -30,5 +31,24 @@ internal static void UpdateText(SourceText newText, ITextBuffer buffer, EditOpti
 
             edit.ApplyAndLogExceptions();
         }
+
+        internal static void UpdateText(IEnumerable<TextChange> textChanges, ITextBuffer buffer, EditOptions options)
+        {
+            using var edit = buffer.CreateEdit(options, reiteratedVersionNumber: null, editTag: null);
+            var oldSnapshot = buffer.CurrentSnapshot;
+            var oldText = oldSnapshot.AsText();
+            if (CodeAnalysis.Workspace.TryGetWorkspace(oldText.Container, out var workspace))
```

Comment:
when would this be false?

Reply:
Modified this a bit to unify the logic between the two edit application methods.  I'm leaving this check as it is an existing check.